cpp/tests/test_odeseir.cpp

```diff
@@ -226,14 +250,8 @@ TEST(TestSeir, interpolate_reproduction_numbers)
 
     mio::TimeSeries<double> result = mio::simulate(t0, tmax, dt, model);
 
-    EXPECT_EQ(model.interpolate_reproduction_numbers(result, -0.5), model.get_reproduction_number(0, result));
-    EXPECT_EQ(
-        model.interpolate_reproduction_numbers(result, result.get_num_time_points() - 0.5),
-        model.get_reproduction_number(
-            result.get_num_time_points() - 1,
-            result)); //Interpolation should return the right respectively leftmost reproduction number if time desired time has distance at most one from these points
-    EXPECT_FALSE(model.interpolate_reproduction_numbers(result, -1.5)); //Test for indices out of range
-    EXPECT_FALSE(model.interpolate_reproduction_numbers(result, result.get_num_time_points() + 0.5));
-    EXPECT_EQ(model.interpolate_reproduction_numbers(result, 1.5).value(), 2.32796973442132285);
-    EXPECT_EQ(model.interpolate_reproduction_numbers(result, 4.75).value(), 1.16201052728287635);
+    EXPECT_FALSE(model.get_reproduction_number(-0.5, result)); //Test for indices out of range
+    EXPECT_FALSE(model.get_reproduction_number(result.get_num_time_points() - 0.5, result));
```

Comment:
Why should num_time_points()-0.5 fail? num_time_points returns an index and -0.5 makes it implicitly a time, so the double/ScalarType version function is called. This does not make sense to me.